crates/core/app/src/params/change.rs

```diff
@@ -71,26 +95,40 @@ impl ChainParameters {
     }
 
     pub fn check_valid(&self) -> Result<()> {
-        let ChainParameters {
-            chain_id,
-            epoch_duration,
-            unbonding_epochs,
-            active_validator_limit,
-            base_reward_rate,
-            slashing_penalty_misbehavior,
-            slashing_penalty_downtime,
-            signed_blocks_window_len,
-            missed_blocks_maximum,
-            ibc_enabled,
-            inbound_ics20_transfers_enabled,
-            outbound_ics20_transfers_enabled,
-            proposal_voting_blocks,
-            proposal_deposit_amount,
-            proposal_valid_quorum,
-            proposal_pass_threshold,
-            proposal_slash_threshold,
-            dao_spend_proposals_enabled: _,
-            // IMPORTANT: Don't use `..` here! We want to ensure every single field is verified!
+        let AppParameters {
+            chain_params:
+                ChainParameters {
+                    chain_id,
+                    epoch_duration,
+                },
+            stake_params:
+                StakeParameters {
+                    unbonding_epochs,
+                    active_validator_limit,
+                    base_reward_rate,
+                    slashing_penalty_misbehavior,
+                    slashing_penalty_downtime,
+                    signed_blocks_window_len,
+                    missed_blocks_maximum,
+                },
+            ibc_params:
+                IBCParameters {
+                    ibc_enabled,
+                    inbound_ics20_transfers_enabled,
+                    outbound_ics20_transfers_enabled,
+                },
+            governance_params:
+                GovernanceParameters {
+                    proposal_voting_blocks,
+                    proposal_deposit_amount,
+                    proposal_valid_quorum,
+                    proposal_pass_threshold,
+                    proposal_slash_threshold,
+                },
+            dao_params:
+                DaoParameters {
+                    dao_spend_proposals_enabled: _,
+                }, // IMPORTANT: Don't use `..` here! We want to ensure every single field is verified!
```

Comment:
Now that we have per-component parameters, it feels like we can have a much better solution to this problem, since we can do all of the parameter validation "locally" in each component, whereas before, our only option was to do all of that logic "at a distance" here in the governance component (cc @plaidfinch)